dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -871,7 +887,30 @@ public Set<Entry> getStarredWorkflows(@ApiParam(hidden = true) @Parameter(hidden
     public Set<Entry> getStarredServices(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User user) {
         User u = userDAO.findById(user.getId());
         checkNotNullUser(u);
-        return u.getStarredEntries().stream().filter(element -> element instanceof Service)
+        return this.getStarredEntries(u, EntryType.SERVICE);
+    }
+
+    /**
+     * Helper function for getStarred endpoints
+     *
+     * @param user
+     * @param entryType
+     * @return a user's starredEntries for desired entryType
+     */
+    public Set<Entry> getStarredEntries(User user, EntryType entryType) {
+        Predicate<Entry> checkInstance;
+
+        if (entryType == EntryType.WORKFLOW) {
+            checkInstance = BioWorkflow.class::isInstance;
+        } else if (entryType == EntryType.SERVICE) {
+            checkInstance = Service.class::isInstance;
+        } else if (entryType == EntryType.NOTEBOOK) {
+            checkInstance = Notebook.class::isInstance;
+        } else { // else would either be Tool or AppTool
+            checkInstance = element -> (element instanceof Tool || element instanceof AppTool);
```

Comment:
Hmmm, this is not a good assumption because like how we're adding notebooks today, we may add something else later which would end up here.  Recommend a check for Tool explicitly for this clause and throw an `UnsupportedOperationException` for an unknown entryType.